sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java

```diff
@@ -636,10 +638,26 @@ public ServiceBusSenderClient buildClient() {
         private String sessionId;
         private String subscriptionName;
         private String topicName;
+        // .Net have default of  5 minutes
+        private Duration maxAutoRenewDuration = MAX_LOCK_RENEW_DEFAULT_DURATION;
 
         private ServiceBusSessionReceiverClientBuilder() {
         }
 
+        /**
+         * Sets the amount of time to continue auto-renewing the session lock. Setting {@link Duration#ZERO} or
+         * {@code null} disables auto-renewal.
+         *
+         * @param maxAutoRenewDuration the amount of time to continue auto-renewing the session lock.
+         * {@link Duration#ZERO} or {@code null} indicates that auto-renewal is disabled.
+         *
+         * @return The updated {@link ServiceBusSessionReceiverClientBuilder} object.
+         */
+        public ServiceBusSessionReceiverClientBuilder mxAutoRenewDuration(Duration maxAutoRenewDuration) {
+            this.maxAutoRenewDuration = maxAutoRenewDuration;
```

Comment:
I think we should move the validation of this argument to here, so we don't need to bloat buildAsyncClient. Same with prefetch as well.

Reply:
I did moved it once but later I saw a test which validating illegal value for `prefetch` thro error at time of buildClient().
Also from java convention point of view, I think We throw error when we people call .build() . 
The flow as I see as user
1. I configure values  in builder
2. I build (This is the place, more appropriate to throw illegal value exception)